File: Day4/D4.java
import java.util.*;

class Day4 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int[] hbg = new int[sc.nextInt()];
        for (int i = 0; i < hbg.length; i++) {
            hbg[i] = sc.nextInt();
        }

        System.out.println(isComplete(hbg));
    }

    static int isComplete(int[] arr) {
        int max = 0, maxIdx = 0, result = 0;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] > max) {
                max = arr[i];
                maxIdx = i;
            }
        }

        for (int j = maxIdx; j < arr.length - 1; j++) {
            if (arr[j] < arr[j + 1])
                return result;
        }
        for (int k = maxIdx; k > 1; k--) {
            if (arr[k] < arr[k - 1])
                return result;
        }

        for (int l : arr)
            result += l;
        return result;
    }
}



/**
코드 해설

1. new int[] 배열 생성, 길이는 Scanner로 입력받음
2. 배열에 값 Scanner로 입력받음
3. isComplete 메소드 실행
	- 배열을 순회하며 최대값이 위치한 인덱스를 구함
	- 최대값을 기준으로 배열 왼쪽과 오른쪽으로 배열 순차 탐색하면서 현재 값과 다음 값을 비교하여 값이 같거나 감소하는지를 확인, 조건을 만족하면 배열의 합 반환, 그렇지 않는 경우 0 반환
4. 결과 출력
 */

File: Week1/D4.py
def isComplete(arr) :
    maxIdx = arr.index(max(arr))
    for i in range(maxIdx, 0, -1) :
        if arr[i] < arr[i - 1] : return 0
    for j in range(maxIdx, len(arr) - 1) :
        if arr[j] < arr[j + 1] : return 0
    return sum(arr)

n = int(input())
hbg = list(map(int, input().split()))[:n]

print(isComplete(hbg))


"""
코드 해설

1. N값 입력 후 map 이용하여 연속으로 값 입력받음, 이후 함수에 사용하기 위해 리스트로 변환
2. isComplete 함수 실행
    - 최댓값을 가지는 인덱스 찾음
    - 최댓값을 기준으로 리스트 왼쪽과 오른쪽으로 순차 탐색하면서 현재 값과 다음 값을 비교하여 값이 같거나 감소하는지 확인, 그렇지 않으면 0 반환
3. 결과값 출력
"""